tests/dynamicconfig.go

```diff
@@ -75,10 +80,27 @@ func (d *dcClient) GetValue(name dynamicconfig.Key) []dynamicconfig.ConstrainedV
 	return d.fallback.GetValue(name)
 }
 
-func (d *dcClient) OverrideValue(name dynamicconfig.Key, value any) {
+// OverrideValue overrides a value for the duration of a test. All values overridden via
+// this API will be cleared when ClearOverrides is called.
+func (d *dcClient) OverrideValue(t *testing.T, name dynamicconfig.Key, value any) {
```

Comment:
I think we can simplify this a lot by removing every override/remove method other than `OverrideValue`, and then resetting to the original value that it was in the cleanup like this:

```go
d.Lock()
defer d.Unlock()
original, ok := d.overrides[name]
d.overrides[name] = value

t.Cleanup(func() {
  if ok {
    d.overrides[name] = value
  } else {
    delete(d.overrides, name)
   }
})
```

That way, there's no reason to distinguish suite vs. test overrides because, when the test exits, it will automatically reset to the value the suite set it to.

Reply:
I'm ok with this and actually had this epiphany (if you can call it that) at the grocery store this evening (I should probably log off).

I'll update it to this behavior as its by far the sanest approach